libcudacxx/include/cuda/std/version

```diff
@@ -86,6 +90,9 @@
 #endif // _CCCL_COMPILER(MSVC) && _CCCL_STD_VER >= 2020
 #define __cccl_lib_null_iterators 201304L
 #define __cccl_lib_optional       202110L
+#ifdef CCCL_ENABLE_OPTIONAL_REF
+#  define __cpp_lib_optional_ref 202602L
```

Comment:
@miscco Should these new feature macros have `__cccl` prefix to avoid clashing with libc++ if/when they implement the extension?